test/test_implicit_ab.py

```diff
@@ -0,0 +1,127 @@
+    from scipy.optimize import root
+    return root(lambda unk: unk - f(t=t, y=sub_y + coeff*unk), guess).x
+
+
+def am_solver_hook(solve_expr, solve_var, solver_id, guess):
+    from dagrt.expression import match, substitute
+
+    pieces = match("unk - <func>rhs(t=t, y=sub_y + coeff*unk)", solve_expr,
+                   pre_match={"unk": solve_var})
+    pieces["guess"] = guess
+    return substitute("<func>solver(t, sub_y, coeff, guess)", pieces)
+
+
+@pytest.mark.parametrize(("method", "expected_order", "static_dt"), [
+    (AdamsMoultonMethodBuilder("y", order, static_dt=static_dt), order, static_dt)
+    for order in [1, 2, 3, 4, 5]
+    for static_dt in [True, False]
+    ] + [
+    (AdamsMoultonMethodBuilder("y", order, hist_length=order+1,
+        static_dt=static_dt), order, static_dt)
+    for order in [1, 2, 3, 4, 5]
+    for static_dt in [True, False]
+    ])
+def test_am_convergence(python_method_impl, method, expected_order, static_dt):
+    plot_solution = False
+    from utils import DefaultProblem
+    problem = DefaultProblem()
+
+    component_id = method.component_id
+    code = method.generate()
+
+    from leap.implicit import replace_AssignImplicit
+    code = replace_AssignImplicit(code, {"solve": am_solver_hook})
+
+    from pytools.convergence import EOCRecorder
+    eocrec = EOCRecorder()
+
+    # default dts
+    dts = 2 ** -np.array(range(4, 7), dtype=np.float64)  # noqa pylint:disable=invalid-unary-operand-type
+
+    for dt in dts:
+        t = problem.t_start
+        y = problem.initial()
+        final_t = problem.t_end
+
+        from functools import partial
+        interp = python_method_impl(code, function_map={
+            "<func>" + component_id: problem,
+            "<func>solver": partial(am_solver, problem),
+            })
+        interp.set_up(t_start=t, dt_start=dt, context={component_id: y})
+
+        times = []
+        values = []
+        for event in interp.run(t_end=final_t):
+            if isinstance(event, interp.StateComputed):
+                assert event.component_id == component_id
+                values.append(event.state_component[0])
+                times.append(event.t)
+
+        assert abs(times[-1] - final_t) / final_t < 0.1
+
+        times = np.array(times)
+
+        if plot_solution:
+            import matplotlib.pyplot as pt
+            pt.plot(times, values, label="comp")
+            pt.plot(times, problem.exact(times), label="true")
+            pt.show()
+
+        error = abs(values[-1] - problem.exact(final_t))
+        eocrec.add_data_point(dt, error)
+
+    print("------------------------------------------------------")
+    print("%s: expected order %d" % (method.__class__.__name__,
+                                     expected_order))
+    print("------------------------------------------------------")
+    print(eocrec.pretty_print())
+
+    orderest = eocrec.estimate_order_of_convergence()[0, 1]
+    assert orderest > expected_order * 0.9
```

Comment:
It looks like this code largely duplicates what's already in `test_imex.py`. I would prefer if this could be expressed in one shared test.